src/go/rdctl/pkg/autostart/autostart_darwin.go

```diff
@@ -0,0 +1,54 @@
+package autostart
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+const launchAgentFileContents = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>Label</key>
+    <string>io.rancherdesktop.autostart</string>
+    <key>ProgramArguments</key>
+    <array>
+        <string>/Applications/Rancher Desktop.app/Contents/MacOS/Rancher Desktop</string>
+    </array>
+    <key>ProcessType</key>
+    <string>Interactive</string>
+    <key>RunAtLoad</key>
+    <true/>
+    <key>KeepAlive</key>
+    <false/>
+</dict>
+</plist>
+`
+
+var launchAgentFilePath string
+
+func init() {
```

Comment:
I'm not a big fan of `init` functions because you can't properly do error reporting.

Any reason this needs to be a global variable? You could just inline the call into the code that uses it.

Reply:
There is an idiom using `sync.Once` to get the effect of `init`, but defer the initialization until the first time you use the value. See https://github.com/lima-vm/lima/blob/21c7e311ba41aa5d0be537bdc98cf0c3bb2c8769/pkg/osutil/machineid.go#L17-L37 for an example.

But for cheap functions, I think it is best not to bother with caching at all. It only becomes worth it if the function is called a million times. 😄